planner: update SPM doc (#16772)
pingcap/docs: sql-plan-management.md

When using this feature, note the following:

- The feature generates hints according to historical execution plans and uses the generated hints for binding. Because historical execution plans are stored in [Statement Summary Tables](/statement-summary-tables.md), before using this feature, you need to enable the [`tidb_enable_stmt_summary`](/system-variables.md#tidb_enable_stmt_summary-new-in-v304) system variable first.

> - For TiFlash queries, Join queries with three or more tables, and queries that contain subqueries, auto-generated hints might not be complete and able to fix the plan completely, for such cases a warning will be outputted.

Reviewer: ```suggestion
- For TiFlash queries, Join queries with three or more tables, and queries that contain subqueries, the auto-generated hints are not complete enough, which might result in the plan not being fully bound. In such cases, a warning will occur when creating a binding.
```

> - For TiFlash queries, Join queries with three or more tables, and queries that contain subqueries, the auto-generated hints are not complete enough, which might result in the plan not being fully bound. In such cases, a warning will occur when creating a binding.

Reviewer 2: ```suggestion
- For TiFlash queries, Join queries with three or more tables, and queries that contain subqueries, the auto-generated hints are not adequate, which might result in the plan not being fully bound. In such cases, a warning will occur when creating a binding.
```

1 row in set (0.00 sec)
```

> ## Utilize Statement Summary table to obtain queries that need to be bound

Reviewer: ```suggestion
## Utilize the Statement Summary table to obtain queries that need to be bound
```

> [Statement Summary](/statement-summary-tables.md) stores recent SQL execution information, such as latency, execution times, corresponding query plans, etc. You can query Statement Summary to get qualified `plan_digest`, and [create bindings according to these historical execution plans](/sql-plan-management.md#create-a-binding-according-to-a-historical-execution-plan).

Reviewer: ```suggestion
[Statement Summary](/statement-summary-tables.md) stores recent SQL execution information, such as latency, execution times, and corresponding query plans. You can query Statement Summary to get qualified `plan_digest`, and then [create bindings according to these historical execution plans](/sql-plan-management.md#create-a-binding-according-to-a-historical-execution-plan).
```

> ## Utilize the Statement Summary table to obtain queries that need to be bound
> 
> [Statement Summary](/statement-summary-tables.md) stores recent SQL execution information, such as latency, execution times, and corresponding query plans. You can query Statement Summary to get qualified `plan_digest`, and then [create bindings according to these historical execution plans](/sql-plan-management.md#create-a-binding-according-to-a-historical-execution-plan).
> 
> The following example searches for SELECT statements that have been executed more than 10 times in the past two weeks, and have multiple execution plans without SQL binding. It sorts the queries by the execution times, and binds the top 100 queries to their fastest plans.

Reviewer 3: ```suggestion
## Utilize the statement summary table to obtain queries that need to be bound

[Statement summary](/statement-summary-tables.md) records recent SQL execution information, such as latency, execution times, and corresponding query plans. You can query statement summary tables to get qualified `plan_digest`, and then [create bindings according to these historical execution plans](/sql-plan-management.md#create-a-binding-according-to-a-historical-execution-plan).

The following example queries `SELECT` statements that have been executed more than 10 times in the past two weeks, and have multiple execution plans without SQL binding. It sorts the queries by the execution times, and binds the top 100 queries to their fastest plans.
```

> The following example searches for SELECT statements that have been executed more than 10 times in the past two weeks, have unstable execution plans, and have not been bound. It sorts the queries by the execution times, and binds the top 100 queries to their fastest plans.

Reviewer 2: ```suggestion
The following example searches for SELECT statements that have been executed more than 10 times in the past two weeks, have multiple execution plans without SQL binding. It sorts the queries by the execution times, and binds the top 100 queries to their fastest plans.
```

  CONCAT('create global binding from history using plan digest "', plan_digest, '"') as binding_stmt 
  FROM stmts t1
  WHERE avg_latency = (SELECT min(avg_latency) FROM stmts t2   -- The plan with the lowest query latency
                       WHERE t2.`digest` = t1.`digest`)
)

SELECT any_value(digest_text) as query, 
       SUM(exec_count) as exec_count, 
       plan_hint, binding_stmt
FROM stmts, best_plans
WHERE stmts.`digest` = best_plans.`digest`
  AND summary_begin_time > DATE_SUB(NOW(), interval 14 day)    -- Executed in the past 2 weeks
  AND stmt_type = 'Select'                                     -- Only consider select statements
  AND schema_name NOT IN ('INFORMATION_SCHEMA', 'mysql')       -- Not an internal query
  AND plan_in_binding = 0                                      -- No binding yet
GROUP BY stmts.`digest`
  HAVING COUNT(DISTINCT(stmts.plan_digest)) > 1                -- This query is unstable. It has more than 1 plan.
         AND SUM(exec_count) > 10                              -- High-frequency, and has been executed more than 10 times.
ORDER BY SUM(exec_count) DESC LIMIT 100;                       -- Top 100 high-frequency queries.
```

> By applying certain filtering conditions to obtain queries that meet the criteria, you can then directly execute the statements corresponding to the `binding_stmt` column to create bindings.

Reviewer: ```suggestion
By applying certain filtering conditions to obtain queries that meet the criteria, you can then directly execute the statements in the corresponding `binding_stmt` column to create bindings.
```
